core/src/main/scala/cats/MonadFilter.scala

```diff
@@ -15,7 +15,4 @@ import simulacrum.typeclass
 
   def filter[A](fa: F[A])(f: A => Boolean): F[A] =
     flatMap(fa)(a => if (f(a)) pure(a) else empty[A])
-
-  def filterM[A](fa: F[A])(f: A => F[Boolean]): F[A] =
-    flatMap(fa)(a => flatMap(f(a))(b => if (b) pure(a) else empty[A]))
```

Comment:
I've removed this based on https://github.com/typelevel/cats/issues/1054#issuecomment-226961198 and to prevent confusion with `filterA`, but that should maybe be a topic for a separate PR.